src/EnergyPlus/ElectricBaseboardRadiator.cc

```diff
@@ -558,29 +528,20 @@ namespace ElectricBaseboardRadiator {
         // PURPOSE OF THIS SUBROUTINE:
         // This subroutine initializes the Baseboard units during simulation.
 
-        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        int ZoneNode;
-
-        auto &ElecBaseboard = state.dataElectBaseboardRad->ElecBaseboard;
+        auto &elecBaseboard = state.dataElectBaseboardRad->ElecBaseboard(BaseboardNum);
         int NumElecBaseboards = state.dataElectBaseboardRad->NumElecBaseboards;
-        auto &ZeroSourceSumHATsurf = state.dataElectBaseboardRad->ZeroSourceSumHATsurf;
-        auto &QBBElecRadSource = state.dataElectBaseboardRad->QBBElecRadSource;
-        auto &QBBElecRadSrcAvg = state.dataElectBaseboardRad->QBBElecRadSrcAvg;
-        auto &LastQBBElecRadSrc = state.dataElectBaseboardRad->LastQBBElecRadSrc;
-        auto &LastSysTimeElapsed = state.dataElectBaseboardRad->LastSysTimeElapsed;
-        auto &LastTimeStepSys = state.dataElectBaseboardRad->LastTimeStepSys;
 
         // Do the one time initializations
         if (state.dataElectBaseboardRad->MyOneTimeFlag) {
             // initialize the environment and sizing flags
             state.dataElectBaseboardRad->MyEnvrnFlag.dimension(NumElecBaseboards, true);
```

Comment:
This SCREAMS that we should just create scalars on the baseboard class:

```
bool MySizeFlag = true;
Real64 QBBElecRadSource = 0.0;
...etc
```

...then we wouldn't have to do this extra allocation, and we wouldn't even need the MyOneTimeFlag **at all** because it's only purpose is to do a one time check to allocate these other things.

The only odd one out is the ZeroSourceSumHATsurf which is allocated to the number of zones, indicating it could be an attribute on the zone data structure.  Worth a thought about the best path forward, just noting it for now.

Reply:
Good suggestion and easy enough to do here.